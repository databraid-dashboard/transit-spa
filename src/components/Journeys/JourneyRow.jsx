import React from 'react';
import PropTypes from 'prop-types';

import { Header } from 'semantic-ui-react';

import TimeToLeave from './Journey/TimeToLeave';
import JourneyVisualization from './Journey/JourneyVisualization';
import ArriveByEstimate from './Journey/ArriveByEstimate';
import CurrentConditionsStatus from './Journey/CurrentConditionsStatus';

import '../../App.css';

const headerStyle = {
  marginTop: '0px',
  marginBottom: '0px',
};

const journeyRowStyle = {
  alignItems: 'center',
  display: 'flex',
  padding: '0.5rem',
  clear: 'both',
  margin: 'auto',
};

const minStyle = {
  padding: 'inherit',
  minWidth: '80px',
};

const JourneyRow = ({ timeToLeaveInSeconds, steps, eta, conditionStatus, callRefreshJourneys }) => (
  <div className="journey-row" style={journeyRowStyle}>
    <div style={minStyle}>
      <p style={headerStyle}>leave in</p>
      <TimeToLeave
        timeToLeaveInSeconds={timeToLeaveInSeconds}
        callRefreshJourneys={callRefreshJourneys}
      />
    </div>

    <div style={minStyle}>
      <JourneyVisualization steps={steps} />
    </div>
    <div style={minStyle}>
      <Header as="h4" style={headerStyle}>
        arrive by
      </Header>
      <ArriveByEstimate eta={eta} />
      <CurrentConditionsStatus conditionStatus={conditionStatus} />
    </div>
  </div>
);

JourneyRow.propTypes = {
  timeToLeaveInSeconds: PropTypes.number.isRequired,
  steps: PropTypes.arrayOf(PropTypes.object).isRequired,
  eta: PropTypes.string.isRequired,
  conditionStatus: PropTypes.string.isRequired,
  callRefreshJourneys: PropTypes.func.isRequired,
};

JourneyRow.defaultProps = {
  timeToLeaveInSeconds: 1,
  steps: [{}],
  eta: '',
  conditionStatus: 'on-time',
  callRefreshJourneys: () => {},
};

export default JourneyRow;
