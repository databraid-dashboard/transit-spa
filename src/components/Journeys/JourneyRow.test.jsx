import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import JourneyRow from './JourneyRow';

describe('JourneyRow', () => {
  it('should render when passed required props', () => {
    const timeToLeaveInSeconds = 120;
    const steps = [
      { duration: '1 mins', instruction: 'Walk to Some St. Station', mode: 'WALKING' },
    ];
    const eta = '11:00am';
    const conditionStatus = 'on-time';

    const component = mount(
      <JourneyRow
        timeToLeaveInSeconds={timeToLeaveInSeconds}
        steps={steps}
        eta={eta}
        conditionStatus={conditionStatus}
        callRefreshJourneys={() => {}}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
