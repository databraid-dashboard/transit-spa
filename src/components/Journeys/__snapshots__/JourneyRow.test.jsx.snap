// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JourneyRow should render when passed required props 1`] = `
<JourneyRow
  callRefreshJourneys={[Function]}
  conditionStatus="on-time"
  eta="11:00am"
  refreshJourneys={[Function]}
  steps={
    Array [
      Object {
        "duration": "1 mins",
        "instruction": "Walk to Some St. Station",
        "mode": "WALKING",
      },
    ]
  }
  timeToLeaveInSeconds={120}
>
  <div
    className="journey-row"
    style={
      Object {
        "alignItems": "center",
        "clear": "both",
        "display": "flex",
        "margin": "auto",
        "padding": "0.5rem",
      }
    }
  >
    <div
      style={
        Object {
          "minWidth": "80px",
          "padding": "inherit",
        }
      }
    >
      <p
        style={
          Object {
            "marginBottom": "0px",
            "marginTop": "0px",
          }
        }
      >
        leave in
      </p>
      <TimeToLeave
        callRefreshJourneys={[Function]}
        refreshJourneys={[Function]}
        timeToLeaveInSeconds={120}
      >
        <Header
          as="h1"
          color="grey"
          style={
            Object {
              "marginBottom": "0rem",
              "marginTop": "0rem",
            }
          }
        >
          <h1
            className="ui grey header"
            style={
              Object {
                "marginBottom": "0rem",
                "marginTop": "0rem",
              }
            }
          >
            <Timer
              onComplete={[Function]}
              seconds={120}
            >
              <div>
                02
                :
                00
              </div>
            </Timer>
          </h1>
        </Header>
      </TimeToLeave>
    </div>
    <div
      style={
        Object {
          "minWidth": "80px",
          "padding": "inherit",
        }
      }
    >
      <JourneyVisualization
        steps={
          Array [
            Object {
              "duration": "1 mins",
              "instruction": "Walk to Some St. Station",
              "mode": "WALKING",
            },
          ]
        }
      >
        <StepGroup
          className="journeyVizColumn"
        >
          <div
            className="ui steps journeyVizColumn"
          >
            <Step>
              <div
                className="step"
                onClick={[Function]}
              >
                <TransitNode
                  step={
                    Object {
                      "duration": "1 mins",
                      "instruction": "Walk to Some St. Station",
                      "mode": "WALKING",
                    }
                  }
                >
                  <Popup
                    on="hover"
                    position="top left"
                    trigger={
                      <div>
                        <Image
                          as="img"
                          src="walk-web-med.png"
                          style={
                            Object {
                              "float": "left",
                              "maxHeight": "22px",
                            }
                          }
                          ui={true}
                        />
                        
                      </div>
                    }
                  >
                    <Portal
                      closeOnDocumentClick={true}
                      closeOnEscape={true}
                      closeOnTriggerMouseLeave={true}
                      mouseEnterDelay={50}
                      mouseLeaveDelay={70}
                      onClose={[Function]}
                      onMount={[Function]}
                      onOpen={[Function]}
                      onUnmount={[Function]}
                      openOnTriggerClick={true}
                      openOnTriggerMouseEnter={true}
                      trigger={
                        <div>
                          <Image
                            as="img"
                            src="walk-web-med.png"
                            style={
                              Object {
                                "float": "left",
                                "maxHeight": "22px",
                              }
                            }
                            ui={true}
                          />
                          
                        </div>
                      }
                    >
                      <div
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                      >
                        <Image
                          as="img"
                          src="walk-web-med.png"
                          style={
                            Object {
                              "float": "left",
                              "maxHeight": "22px",
                            }
                          }
                          ui={true}
                        >
                          <img
                            className="ui image"
                            src="walk-web-med.png"
                            style={
                              Object {
                                "float": "left",
                                "maxHeight": "22px",
                              }
                            }
                          />
                        </Image>
                      </div>
                    </Portal>
                  </Popup>
                </TransitNode>
              </div>
            </Step>
          </div>
        </StepGroup>
      </JourneyVisualization>
    </div>
    <div
      style={
        Object {
          "minWidth": "80px",
          "padding": "inherit",
        }
      }
    >
      <Header
        as="h4"
        style={
          Object {
            "marginBottom": "0px",
            "marginTop": "0px",
          }
        }
      >
        <h4
          className="ui header"
          style={
            Object {
              "marginBottom": "0px",
              "marginTop": "0px",
            }
          }
        >
          arrive by
        </h4>
      </Header>
      <ArriveByEstimate
        eta="11:00am"
      >
        <Header
          as="h4"
          style={
            Object {
              "marginBottom": "0rem",
              "marginTop": "0rem",
            }
          }
        >
          <h4
            className="ui header"
            style={
              Object {
                "marginBottom": "0rem",
                "marginTop": "0rem",
              }
            }
          >
            11:00am
          </h4>
        </Header>
      </ArriveByEstimate>
      <CurrentConditionsStatus
        conditionStatus="on-time"
      >
        <Popup
          content="on-time"
          inverted={true}
          offset={50}
          on="hover"
          position="right center"
          style={
            Object {
              "borderRadius": 0,
              "opacity": 0.7,
              "padding": "2em",
            }
          }
          trigger={
            <Label
              color="green"
              style={
                Object {
                  "marginBottom": "0rem",
                  "marginTop": "0.5rem",
                }
              }
            >
              on-time
            </Label>
          }
        >
          <Portal
            closeOnDocumentClick={true}
            closeOnEscape={true}
            closeOnTriggerMouseLeave={true}
            mouseEnterDelay={50}
            mouseLeaveDelay={70}
            onClose={[Function]}
            onMount={[Function]}
            onOpen={[Function]}
            onUnmount={[Function]}
            openOnTriggerClick={true}
            openOnTriggerMouseEnter={true}
            trigger={
              <Label
                color="green"
                style={
                  Object {
                    "marginBottom": "0rem",
                    "marginTop": "0.5rem",
                  }
                }
              >
                on-time
              </Label>
            }
          >
            <Label
              color="green"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "marginBottom": "0rem",
                  "marginTop": "0.5rem",
                }
              }
            >
              <div
                className="ui green label"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "marginBottom": "0rem",
                    "marginTop": "0.5rem",
                  }
                }
              >
                on-time
              </div>
            </Label>
          </Portal>
        </Popup>
      </CurrentConditionsStatus>
    </div>
  </div>
</JourneyRow>
`;
